### PR TITLE
re construction of argument/var commands

### PR DESCRIPTION
**Pull request by oddcoder**
afa for base pointer arguments/vars
afe for stack pointet arguments/vaes
afv for registers
fixing the J and \0 suffix
using the right register indexer
fixng the negative offsets of afa
fixing the afa afe afv views of pdf
fixing errors in aa
fixing the stack bases arg/var
fixing the varsub
fixing afa*
fixing afat
making afx- space neutral
enabling register renaming at varsub
updating the built-in help

### CODE REVIEW

**Comment by oddcoder** [comment]
FIX #4946

**Comment by oddcoder** [comment]
It will break some tests but I fixed lots of them on (if not all) on local machine

File: libr/anal/var.c [hunk 1]
```diff
@@ -491,36 +539,42 @@ R_API void r_anal_var_list_show(RAnal *anal, RAnalFunction *fcn, int kind, int m
...
+				anal->cb_printf ("reg %s %s @ %s\n",
+					var->type, var->name, i->name);
 				}
 				break;
 			case 'e':
-				anal->cb_printf ("var %s %s @ %s+0x%x\n",
-					var->type, var->name,
-					anal->reg->name[R_REG_NAME_SP],
-					var->delta);
+				if ( var->delta < fcn->stack) {
```
Comment (radare) [inline libr/anal/var.c]: Drop that spce after (

File: libr/anal/var.c [hunk 2]
```diff
@@ -491,36 +539,42 @@ R_API void r_anal_var_list_show(RAnal *anal, RAnalFunction *fcn, int kind, int m
...
+					anal->cb_printf ("var %s %s @ %s+0x%x\n",
+						var->type, var->name,
```
Comment (radare) [inline libr/anal/var.c]: Looks like dupe code only changing arg and var

File: libr/core/disasm.c [hunk 3]
```diff
@@ -1008,46 +1008,48 @@ static void ds_show_functions(RDisasmState *ds) {
...
 				}
 				break;
 			case 'e':
-				r_cons_printf ("var %s %s @ %s+0x%x",
-					var->type, var->name,
-					core->anal->reg->name[R_REG_NAME_SP],
-					var->delta);
+				if ( var->delta < f->stack) {
+					r_cons_printf ("var %s %s @ %s+0x%x",
+						var->type, var->name,
```
Comment (radare) [inline libr/core/disasm.c]: Dupe

**Comment by radare** [comment]
Can you update broken tests?

**Comment by Maijin** [comment]
Doesn't build on OSX

**Comment by oddcoder** [comment]
now it builds everywhere

File: libr/anal/var.c [hunk 4]
```diff
@@ -441,48 +476,61 @@ R_API void r_anal_var_list_show(RAnal *anal, RAnalFunction *fcn, int kind, int m
 		switch (mode) {
 		case '*':
 			// we cant express all type info here :(
-			anal->cb_printf ("af%c %d %s %s @ 0x%"PFMT64x"\n",
-				kind, var->delta,
-				var->name, var->type, fcn->addr);
+			if (kind == 'v') { //registers
+				RRegItem *i = r_reg_index_get (anal->reg, var->delta);
+				if (!i) {
+					eprintf("Register not found");
```
Comment (radare) [inline libr/anal/var.c]: missing space before (

Comment (radare) [inline libr/anal/var.c]: maybe add some more verbosity, like which reg is trying to resolve

File: libr/anal/var.c [hunk 5]
```diff
@@ -441,48 +476,61 @@ R_API void r_anal_var_list_show(RAnal *anal, RAnalFunction *fcn, int kind, int m
...
+						var->name, var->type,anal->reg->name[R_REG_NAME_BP],
+						-var->delta);
+
+				}
 				break;
-			case 'v':
+			case 'v':{
+				RRegItem *i = r_reg_index_get (anal->reg, var->delta);
+				if (!i) {
+					eprintf("Register not found");
```
Comment (radare) [inline libr/anal/var.c]: same as above .dupe code?

**Comment by radare** [comment]
lgtm, but r2r needs to be updated

**Comment by oddcoder** [comment]
update here
https://github.com/radare/radare2-regressions/pull/442

**Pull request closed without merging**